Software/lib/GNSE_HAL/GNSE_hal.c

```diff
@@ -0,0 +1,240 @@
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to initialize LIS2DH12 Accelerometer \r\n");
+#endif
+    }
+
+    if (SHTC3_probe() != SHTC3_STATUS_OK)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to initialize SHTC3 Temperature Sensor \r\n");
+#endif
+    }
+}
+
+/**
+  * @brief Deinitialises hardware of the Flash IC
+  * @param none
+  * @return none
+  */
+void GNSE_Flash_DeInit(void)
+{
+    if (GNSE_BSP_LS_Off(LOAD_SWITCH_FLASH) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Flash Load Switch \r\n");
+#endif
+    }
+    if (GNSE_BSP_LS_DeInit(LOAD_SWITCH_FLASH) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Flash Load Switch \r\n");
+#endif
+    }
+    if (GNSE_BSP_Flash_SPI_DeInit() != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Flash SPI \r\n");
+#endif
+    }
+}
+
+/**
+  * @brief Initialises hardware of the Flash IC
+  * @param none
+  * @return none
+  */
+void GNSE_Flash_Init(void)
+{
+    if (GNSE_BSP_LS_Init(LOAD_SWITCH_FLASH) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to initialize Flash Load Switch \r\n");
+#endif
+    }
+    if (GNSE_BSP_LS_On(LOAD_SWITCH_FLASH) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to initialize Flash Load Switch \r\n");
+#endif
+    }
+    if (GNSE_BSP_Flash_SPI_Init() != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to initialize Flash SPI \r\n");
+#endif
+    }
+}
+
+/**
+  * @brief Deinitialises all common GNSE hardware
+  * @param none
+  * @return none
+  */
+void GNSE_DeInit(void)
+{
+    GNSE_Sensors_DeInit();
+    if (GNSE_BSP_Sensor_I2C1_DeInit() != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialise I2C \r\n");
+#endif
+    }
+    GNSE_Flash_DeInit();
+    if (GNSE_BM_DeInit() != BM_OP_SUCCESS)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialise battery \r\n");
+#endif
+    }
+    if (GNSE_BSP_LED_DeInit(LED_BLUE) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Blue LED \r\n");
+#endif
+    }
+    if (GNSE_BSP_LED_DeInit(LED_RED) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Red LED \r\n");
+#endif
+    }
+    if (GNSE_BSP_LED_DeInit(LED_GREEN) != GNSE_BSP_ERROR_NONE)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Green LED \r\n");
+#endif
+    }
+    if (BUZZER_DeInit() != BUZZER_OP_SUCCESS)
+    {
+#if (APP_LOG_ENABLED)
+        APP_PPRINTF("\r\n Failed to de-initialize Buzzer \r\n");
+#endif
+    }
+}
+
+/**
+  * @brief Initialises all common GNSE hardware
+  * @param none
+  * @return none
+  */
+void GNSE_Init(void)
```

Comment:
This function can be more modular, meaning you should be able to pass which sensors/component you want to init.
Please provide a proposal in the issue before implementing and we can discuss how it should be implemented

Reply:
Let me know if you like the struct solution or have some other preferred method of setting configurations.